spec/controllers/shoutouts_controller_spec.rb

```diff
@@ -0,0 +1,245 @@
+                                                                             rich_text: '@Person1 thanks',
+                                                                             recipients: [user5.id.to_s, user3.id.to_s, user4.id.to_s] }) })
+    end
+    let!(:recipients2) do
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user5[:id] })
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user3[:id] })
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user4[:id] })
+    end
+    context 'with valid attributes' do
+      it 'updates shoutouts' do
+        allow(controller).to receive(:current_user).and_return(user)
+        patch :update, params: {  id: 1,
+                                  shoutout: { user_id: user.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '@Person1 thanks',
+                                              recipients: [user2.id, user3.id, user5.id] } }
+        shoutout.reload
+        expect(shoutout.rich_text).to eq('@Person1 thanks')
+        expect(ShoutoutRecipient.all.length).to be(6)
+        expect(ShoutoutRecipient.find_by(user_id: user5.id)[:user_id]).to be(user5.id)
+      end
+    end
+    context 'with similar rich text' do
+      it 'should not update when we have the similar record ' do
+        allow(controller).to receive(:current_user).and_return(user)
+        post :create, params: { shoutout: { user_id: user.id,
+                                            time_period_id: time_period.id,
+                                            rich_text: '@Person1 thanks',
+                                            recipients: [user2.id, user3.id, user4.id] } }
+        expect(ShoutoutRecipient.all.length).to be(9)
+        patch :update, params: {  id: shoutout.id,
+                                  shoutout: { user_id: user.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '@Person1 thanks',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        shoutout.reload
+        expect(shoutout.rich_text).to eq('<br>')
+      end
+      it 'should update when we have a similar rich text but other user and
+          should not update when we have the similar rich text in the same user' do
+        allow(controller).to receive(:current_user).and_return(user1)
+        expect do
+          post :create, params: { shoutout: { user_id: user1.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '@Person1',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+          shoutout.reload
+        end.to change(Shoutout, :count).by(1)
+        expect(ShoutoutRecipient.all.length).to be(9)
```

Comment:
the same, you create new shoutout so please check ShoutoutRecipient connected with this shoutout

Reply:
Thanks I changed